plugins/in_tail/tail_file.c

```diff
@@ -1399,6 +1405,14 @@ int flb_tail_file_chunk(struct flb_tail_file *file)
         capacity = (file->buf_size - file->buf_len) - 1;
     }
 
+    #ifdef __linux__
+    if (ctx->file_cache_advise) {
+        if (posix_fadvise(file->fd, 0, 0, POSIX_FADV_DONTNEED) == -1) {
+            flb_errno();
+            flb_plg_error(ctx->ins, "error during posix_fadvise");
```

Comment:
Maybe this could use [flb_errno_print](https://github.com/fluent/fluent-bit/blob/9652b0dc6fd9e59ef87c3bdc59c89da9abed995b/src/flb_log.c#L673) so we can see the errno?

Reply:
Sure, I'll change that

Reply:
`flb_errno` is just a macro substitution for `flb_errno_print`:
https://github.com/fluent/fluent-bit/blob/9652b0dc6fd9e59ef87c3bdc59c89da9abed995b/include/fluent-bit/flb_log.h#L232
I cannot see the `flb_errno_print` function used anywhere in the code. I will leave it like it is, as also in the same `tail_file.c` it is done exactly the same way as I did: 
https://github.com/fluent/fluent-bit/blob/9652b0dc6fd9e59ef87c3bdc59c89da9abed995b/plugins/in_tail/tail_file.c#L938-L943

Reply:
`flb_errno()` and immediately calling flb_plg_error should be OK.

Reply:
From the posix_fadvice man page, without calling `[fsync(2)](https://man7.org/linux/man-pages/man2/fsync.2.html)` or `[fdatasync(2)](https://man7.org/linux/man-pages/man2/fdatasync.2.html)`, releasing of caches is not guaranteed.

> The implementation may attempt to write back dirty pages
              in the specified region, but this is not guaranteed.  Any
              unwritten dirty pages will not be freed.  If the
              application wishes to ensure that dirty pages will be
              released, it should call [fsync(2)](https://man7.org/linux/man-pages/man2/fsync.2.html) or [fdatasync(2)](https://man7.org/linux/man-pages/man2/fdatasync.2.html) first.

ref: https://man7.org/linux/man-pages/man2/posix_fadvise.2.html

is this intended behavior?

Reply:
For just advice to reduce cache usage, it should be enough.

Reply:
Yes, that is correct. There is no guarantee, it is only an advise. However, it should reduce the cache usage. (And actually does it in my tests)

Reply:
> From the posix_fadvice man page, without calling [fsync(2)](https://man7.org/linux/man-pages/man2/fsync.2.html) or [fdatasync(2)](https://man7.org/linux/man-pages/man2/fdatasync.2.html), releasing of caches is not guaranteed.
> 
> > The implementation may attempt to write back dirty pages
> > in the specified region, but this is not guaranteed.  Any
> > unwritten dirty pages will not be freed.  If the
> > application wishes to ensure that dirty pages will be
> > released, it should call [fsync(2)](https://man7.org/linux/man-pages/man2/fsync.2.html) or [fdatasync(2)](https://man7.org/linux/man-pages/man2/fdatasync.2.html) first.
> 
> ref: https://man7.org/linux/man-pages/man2/posix_fadvise.2.html
> 
> is this intended behavior?

As we are only opening files for read, not for write, there shouldn't be any dirty pages. In our case we use every read page once and after that we don't need the page anymore, so we do not need the page cache at all (so set to POSIX_FADV_DONTNEED).

Reply:
Ah, OK. Got it. Thanks for the explanation.

Reply:
Maybe this read mode should even be the default, but I suggest to first leave it as default `Off`

Reply:
Yup. This is likely affected for resource limited k8s containers or something similar VMs.